### PR TITLE
[properties] Extending PCIInfo struct with product ID field

### PR DESCRIPTION
**Pull request by csoka**
### Details:
- PCIInfo structure's device ID field was previously falsely believed to refer to product ID (for identification of device model), but it actually stores pci endpoint ID of the device in the system, therefore the implemented feature does not address the feature-request (PCIInfo to contain product id) PR #24111 
- This PR extends PCIInfo structure with additional field to include product ID as well

### Tickets:
 - [*CVS-138767*](https://jira.devtools.intel.com/browse/CVS-138767)

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp [hunk 1]
```diff
@@ -171,11 +171,12 @@ void regmodule_properties(py::module m) {
 
     // Special case: ov::device::PCIInfo
     py::class_<ov::device::PCIInfo, std::shared_ptr<ov::device::PCIInfo>> cls_pciinfo(m_device, "PCIInfo");
-    cls_pciinfo.def(py::init<const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&>());
+    cls_pciinfo.def(py::init<const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&>());
```
Comment (p-wysocki) [inline src/bindings/python/src/pyopenvino/core/properties/properties.cpp:174]: Won't adding a new `init` argument with no default value break existing scripts? PyAPI needs to follow strict backwards compatibility rules. Is it possible to have a placeholder value here?

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp [hunk 2]
```diff
@@ -171,11 +171,12 @@ void regmodule_properties(py::module m) {
...
     cls_pciinfo.def_readonly("domain", &ov::device::PCIInfo::domain);
     cls_pciinfo.def_readonly("bus", &ov::device::PCIInfo::bus);
     cls_pciinfo.def_readonly("device", &ov::device::PCIInfo::device);
     cls_pciinfo.def_readonly("function", &ov::device::PCIInfo::function);
+    cls_pciinfo.def_readonly("function", &ov::device::PCIInfo::product);
```
Comment (t-jankowski) [inline src/bindings/python/src/pyopenvino/core/properties/properties.cpp:179]:
```suggestion
    cls_pciinfo.def_readonly("product", &ov::device::PCIInfo::product);
```

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp (hunk 1 above)
Comment (p-wysocki) [reply src/bindings/python/src/pyopenvino/core/properties/properties.cpp:174]: True, I just wanted to make sure we're aware about this. :)

File: src/inference/include/openvino/runtime/properties.hpp [hunk 3]
```diff
@@ -1069,18 +1069,23 @@ struct PCIInfo {
      * @brief PCI function ID
      */
     uint32_t function;
+    /**
+     * @brief PCI device's product ID
+     */
+    uint32_t product;
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:1075]: @vladimir-paramuzov is this field applicable for GPU?

**Pull request closed without merging**